### ISSUE TITLE
UploadedTo Download limit exceeded

### ISSUE THREAD

**Issue by nomad71** [pyload/pyload#1477]
With UploadedTo download limits are now handled as error/warning with debug report:

```
09.06.2015 18:22:19 WARNING   UploadedTo[1444]: You have reached the max. number of possible free downloads for this hour
09.06.2015 18:22:19 WARNING   Download failed: b5s106.part2.rar | Download limit exceeded
09.06.2015 18:22:19 INFO      Debug Report written to debug_UploadedTo_09-06-2015_18-22-19.zip
```

Uploaded has a watining time between downloads for free loading. Pyload used to wait this time ( 3 h if i recall correctly) before retry. Now it just burns captchacredits and gives me debug zips in exchange :P

**Comment by vuolter** [comment]
Please, post the debug_UploadedTo_09-06-2015_18-22-19.zip

**Comment by nomad71** [comment]
Here is the latest example i got:

http://www41.zippyshare.com/v/Vm7S3ecy/file.html

**Comment by ignorancez** [comment]
I am also getting this

**Comment by forenuser** [comment]
I am also getting this. Moreover reconnect doesn't work. Possibly the Uploaded.to plugin needs to be changed for the usage of free downloads.

**Comment by 1337sup3rh4x0r** [comment]
I am seeing the same:

    5195	02.07.2015 17:43:16	WARNING	UploadedTo[3]: You have reached the max. number of possible free downloads for this hour
    5196	02.07.2015 17:43:16	WARNING	Download failed: filexy.rar | Download limit exceeded 

Usually the plugin used to go into waiting status after this but now it just fails.

**Comment by 1337sup3rh4x0r** [comment]
After digging through the code a bit (without any python knowledge) I think that the error is within the checkErrors function of SimpleHoster. Somehow no wait time is defined.

shouldn't this part here (lines 583 to 587) lead to the function recognizing the word "hour" and setting the wait_time to 3600 seconds? Or does this only work when there is a number in front of "hour"?

                if re.search('da(il)?y|today', errmsg, re.I):
                    wait_time = secondsToMidnight(gmt=2)
                else:
                    wait_time = sum(int(v) * {"hr": 3600, "hour": 3600, "min": 60, "sec": 1, "": 1}[u.lower()] for v, u in
                                re.findall(r'(\d+)\s*(hr|hour|min|sec|)', errmsg, re.I))

**Comment by 1337sup3rh4x0r** [comment]
I don't know if this really fixes the issue, but I added some code to the above mentioned lines:

                if re.search('da(il)?y|today', errmsg, re.I):
                    wait_time = secondsToMidnight(gmt=2)
                elif re.search('this hour', errmsg, re.I):
                    wait_time = 3600
                else:
                    wait_time = sum(int(v) * {"hr": 3600, "hour": 3600, "min": 60, "sec": 1, "": 1}[u.lower()] for v, u in
                                re.findall(r'(\d+)\s*(hr|hour|min|sec|)', errmsg, re.I))

So now at least for my uploaded.to downloads this results in the file being queued for one hour and then retried. I hope this doesn't break anything for other hosters. I guess we'll have to wait on @vuolter to make a real fix for this.

**Comment by 1337sup3rh4x0r** [comment]
Obviously the sentence ``You have reached the max. number of possible free downloads for this hour`` is not correctly reflecting the restrictions of a free account.
The website states the restriction ``a file every 180 minutes``. So the correct insert would be:

            elif re.search('this hour', errmsg, re.I):
                wait_time = 10800

This also explains why I am still getting the same message after the wait time has ended :)

**Comment by forenuser** [comment]
OK, wait_time is now set to 3h. In case wait_time > 300 reconnect should be executed. But this doesn't work. I don't have any reconnect stuff in the debug log.

**Comment by vuolter** [comment]
@forenuser Recconnect routine has been just revised, give it a try when you can.
@1337sup3rh4x0r Cheking... anyway check_errors is the right place. Good eye.

**Comment by vuolter** [comment]
https://github.com/pyload/pyload/commit/67a21410eeb8a2bd7bc4d71f7fe5cd68312b702b

**Comment by 1337sup3rh4x0r** [comment]
Thanks for the update! I will check if it works this weekend.

The way I und er stand your fix is that parse_time will return a wait time of 3600s, even though I discovered that uploadednet really let's us wait 10080s. I guess there is no way to read this from the message...
But will the wait timer then simply restart 3 times until it is successful (would be nice) or will it fail before that?
Is there maybe a way to "hardcode" a wait time for certain holsters that don't have meaningful messages?

**Comment by vuolter** [comment]
@1337sup3rh4x0r The plugin will restart 3 time. u.u
A way to get 10080s only for this hoster would be overwrite its check_errors method, but I not recommend this, because the inherited method is updated frequently... and there will be also a reason why it happens. ;)

**Comment by joberreiter** [comment]
Would it help to add the lines
```Python
#: Set exception to the wait time
if re.search(self.DL_LIMIT_PATTERN, self.html, re.I)
    self.set_wait(10800, True)
```
after the call to `self.check_errors()` at `handle_free()` in hoster/UploadedTo.py?

**Comment by forenuser** [comment]
I am based on actual version of https://codeload.github.com/pyload/pyload/zip/stable. I exchanged the files which are available from https://github.com/pyload/pyload/commit/67a21410eeb8a2bd7bc4d71f7fe5cd68312b702b. But it doesn't work. 

216	03.10.2015 10:00:27	CRITICAL	'NoneType' object has no attribute 'UploadedTo'
217	03.10.2015 10:00:29	ERROR	Error importing UploadedTo: No module named Plugin

How can I test if reconnect works?

**Comment by forenuser** [comment]
In case a wrong Captcha is used the download is skipped.
```
03.10.2015 12:49:52 DEBUG     HOSTER UploadedTo[103]: Config option `fallback_premium` not found, use default `True`
03.10.2015 12:49:52 WARNING   Download failed: xxx.S08E03.part1.rar | Parse error: Free download link not found | Plugin may be out of date
03.10.2015 12:49:52 INFO      ADDON UnSkipOnFail: Looking for skipped duplicates of: Dexter.S08E03.720p.BD-iNTENTiON.part1.rar (pid:3)
03.10.2015 12:49:52 INFO      ADDON UnSkipOnFail: No duplicates found
Traceback (most recent call last):
  File "/usr/share/pyload/module/database/DatabaseBackend.py", line 96, in processJob
    self.result = self.f(*self.args, **self.kwargs)
  File "/usr/share/pyload/module/database/FileDatabase.py", line 767, in updateLink
    self.c.execute('UPDATE links SET url=?,name=?,size=?,status=?,error=?,package=? WHERE id=?', (f.url, f.name, f.size, f.status, f.error, str(f.packageid), str(f.id)))
InterfaceError: Error binding parameter 4 - probably unsupported type.
Database Error @ updateLink (PyFile 103: Dexter.S08E03.720p.BD-iNTENTiON.part1.rar@UploadedTo,) {} Error binding parameter 4 - probably unsupported type.
03.10.2015 12:49:53 INFO      Download starts: http://xxx.S08E03.part2.rar
```
